tests/run/interpolation-opt.scala

```diff
@@ -0,0 +1,23 @@
+object Test extends App {
+
+  val one = 1
+  val two = "two"
+  val three = 3.0
+
+  // Test escaping
+  println(raw"$one plus $two\nis $three")
+  println(s"$one plus $two\nis $three")
+
+  // Test empty strings between elements
+  println(raw"a$one$two${three}b")
+  println(s"a$one$two${three}b")
+
+  // Test empty string interpolators
+  println(raw"")
+  println(s"")
+
+  // Make sure that StringContext still works with idents
+  val foo = "Hello"
+  val bar = "World"
+  println(StringContext(foo, bar).s(" "))
```

Comment:
Add a test for:
```scala
def myStringContext= { println("Side effect!"); StringContext }
println(myStringContext(foo, bar).s(" ")) // this shouldn't be optimised away
```
I am not sure it is handled by your implementation

Reply:
Added the test and looks like it handles it correctly.

Reply:
Sorry, my example was erroneous. What I had is mind was:
```scala
println(myStringContext("Foo", "Bar").s(" ")) // this shouldn't be optimised away

```

Reply:
This one too, should not be optimised:
```scala
println({ println("Side effect n2!"); StringContext }.apply("Titi", "Toto").s(" ")) // this shouldn't be optimised away
```

Reply:
Added tests and a fix.